core/src/test/java/org/kohsuke/stapler/framework/io/LargeTextTest.java

```diff
@@ -53,13 +64,54 @@ public void writeLogTo() throws Exception {
             // right
         }
     }
+
     String tail(String text, long start) throws IOException {
-        ByteBuffer bb = new ByteBuffer();
-        bb.write(text.getBytes(), 0, text.length());
-        LargeText t = new LargeText(bb, true);
+        LargeText t;
+        try (ByteBuffer bb = new ByteBuffer()) {
+            bb.write(text.getBytes(), 0, text.length());
+
+            t = new LargeText(bb, true);
+        }
         ByteArrayOutputStream baos = new ByteArrayOutputStream();
         assertEquals(text.length(), t.writeLogTo(start, baos));
         return baos.toString();
     }
 
+    @Issue("#141")
+    @Test
+    @Ignore
+    public void writeLogToWithLargeFile() throws Exception {
+        Path path = Files.createTempFile("stapler-test", ".log.gz");
+        long size = Integer.MAX_VALUE + 256L;
+        writeLargeFile(path, size);
+
+        LargeText t = new LargeText(path.toFile(), StandardCharsets.US_ASCII, true);
+
+        try (OutputStream os = new NullOutputStream()) {
+            long writenCount = t.writeLogTo(0, os);
+
+            assertEquals(size, writenCount);
+        }
+
+        Files.delete(path);
+    }
+
+    private void writeLargeFile(Path path, long size) {
+        // Write the same data over and over again, so the bytes written is high, but the file is
+        // actually very small
+        int chunkSize = 1024;
+        byte[] bytesChunk = Strings.repeat("0", chunkSize).getBytes(StandardCharsets.US_ASCII);
+        try (OutputStream stream = new FileOutputStream(path.toFile())) {
+            long remaining = size;
+            while (remaining > chunkSize) {
+                stream.write(bytesChunk);
+                remaining -= chunkSize;
+            }
+            stream.write(bytesChunk, 0, (int) remaining);
+            stream.flush();
+        } catch (IOException e) {
+            fail(e.getMessage());
```

Comment:
Note that you can just let the method throw up the exception and the test will fail just the same.